### PR TITLE
Move `FrontController::updateQueryString` to `Tools` to make reusable

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Move `FrontController::updateQueryString` to `Tools` and make reusable.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Test green :)
| Fixed issue or discussion?     | N/A
| Related PRs       | N/A
| Sponsor company   | Evolutive

### CODE REVIEW

**Review by FabienPapet — CHANGES_REQUESTED** [review]
Hey there ! 👋 
I don't think moving code from FrontController to Tools helps the migration but I'm not against it.

If you want to do so, I think you should deprecate the old function as well, so it can be removed in next versions. As well as updating all the code using this function inside the project.

**Comment by PululuK** [comment]
> Hey there ! 👋 I don't think moving code from FrontController to Tools helps the migration but I'm not against it.
> 
> If you want to do so, I think you should deprecate the old function as well, so it can be removed in next versions. As well as updating all the code using this function inside the project.

is a protected method, We need realy to add deprecated flag ?

**Comment by FabienPapet** [comment]
Absolutely, because a child class can use it. This is covered by the BC promise. Only the private methods don't need it :)

**Comment by PululuK** [comment]
> 

Oh yeah ! Done

**Review by kpodemski — DISMISSED** [review]

**Comment by FabienPapet** [comment]
Some usages still need to be replaced, and then all good for me

**Review by M0rgan01 — DISMISSED** [review]

**Review by FabienPapet — CHANGES_REQUESTED** [review]
Requesting changes because of the bot

**Comment by kpodemski** [comment]
@FabienPapet, where do you see other usages in the core? unless you meant the Faceted Search module? (which should not block the PR)

**Comment by FabienPapet** [comment]
image

**Review by FabienPapet — APPROVED** [review]

**Review by matks — APPROVED** [review]

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1842,48 +1842,11 @@ public function getCanonicalURL()
...
 
     /**
-     * Generate a URL corresponding to the current page but
-     * with the query string altered.
-     *
-     * If $extraParams is set to NULL, then all query params are stripped.
-     *
-     * Otherwise, params from $extraParams that have a null value are stripped,
-     * and other params are added. Params not in $extraParams are unchanged.
+     * @deprecated Since 9.0 and will be removed in the next major.
```
Comment (boherm) [inline classes/controller/FrontController.php:1845]: Maybe we can add in the comment the new way to use this function. What do you think?

**Review by boherm — COMMENTED** [review]
Ok for me! But just a little comment :)

**Review by matthieu-rolland — APPROVED** [review]
Do you want to apply @boherm 's feedback or should we merge ?

File: classes/controller/FrontController.php (hunk 1 above)
Comment (PululuK) [reply classes/controller/FrontController.php:1845]: the phpdoc in the method description should be enough

**Comment by boherm** [comment]
@matthieu-rolland I'm not blocking this PR, so we can merge as is. ;)

**Review by boherm — APPROVED** [review]

**Review by M0rgan01 — APPROVED** [review]

**Comment by M0rgan01** [comment]
After some checks, it's good for me :) Thanks

**Comment by matks** [comment]
@M0rgan01 don't forget the milestone

**Pull request merged**